sklearn/preprocessing/_target_encoder.py

```diff
@@ -217,12 +240,13 @@ def fit_transform(self, X, y):
 
         Returns
         -------
-        X_trans : ndarray of shape (n_samples, n_features)
+        X_trans : ndarray of shape (n_samples, n_features) or \
+                    (n_samples, (n_features * n_classes))
             Transformed input.
         """
         from ..model_selection import KFold, StratifiedKFold  # avoid circular import
 
-        X_ordinal, X_known_mask, y, n_categories = self._fit_encodings_all(X, y)
+        X_ordinal, X_known_mask, y_encoded, n_categories = self._fit_encodings_all(X, y)
```

Comment:
I like this renaming to `y_encoded`, can this be done in seperate smaller PR?

Reply:
I think it would be easier to keep this, it was changed partly because I now use both original `y` and encoded `y_encoded`